src/net_processing.cpp

```diff
@@ -2057,6 +2078,16 @@ void PeerManagerImpl::UpdatedBlockTip(const CBlockIndex *pindexNew, const CBlock
 {
     SetBestHeight(pindexNew->nHeight);
 
+    // Update internal sync state.
+    // Avoid using 'fInitialDownload' alone since the flag currently cannot go backwards after passing IBD once.
```

Comment:
I'm wondering, does it even make sense to keep such flag at this point? What it would take to adjust all code to going back-and-forth w.r.t. superstale?

Reply:
> I'm wondering, does it even make sense to keep such flag at this point? What it would take to adjust all code to going back-and-forth w.r.t. superstale?

I think that would be an oxymoron. We would be adding code to put the node back into IBD state, because of a stale tip check, when the node is updating the tip.

The `fInitialDownload` flag usage is just a speedup. We know for sure that when the flag is true, it is because the node is in IBD. So we don't need to check anything else in such case.

Aside from that, about the adjustments: the stale tip check uses the last tip updated time (`m_last_tip_update`) which is updated on every new block `BlockConnected` (which is executed prior to `UpdatedBlockTip`). So that would need to be moved to `UpdatedBlockTip`. Also, and more importantly for me, because the stale tip check checks the in-flight block requests map, it requires `cs_main` lock. Which I would try to avoid as much as possible as it usage slows down the net messages processing and validation thread.
A little note about this last point; with something like #27837, we could guard `mapBlocksInFlight` under a different mutex and not `cs_main`.